### PR TITLE
[4.0] [com_content] Parameterized queries (frontend)

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds parameterized queries to `com_content` frontend.

### Testing Instructions

View some articles in frontend.
View and filter archived articles.
Vote on articles.
Create some menu items for archived articles and article lists with different ordering options and view them in frontend.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by Quy** [comment]
Create `Archived Articles` menu item.
Go to the frontend and submit a filter.

> PHP Warning:  mysqli_stmt::bind_param(): Number of variables doesn't match number of parameters in prepared statement in \libraries\vendor\joomla\database\src\Mysqli\MysqliStatement.php on line 420
PHP Warning:  Invalid argument supplied for foreach() in \components\com_content\Model\ArticlesModel.php on line 638

**Comment by SharkyKZ** [comment]
@Quy Try again please.

File: components/com_content/Model/ArticleModel.php [hunk 1]
```diff
@@ -119,50 +166,53 @@ public function getItem($pk = null)
...
-
-					$nowDate = $db->quote($date->toSql());
-
-					$query->where('(a.publish_up IS NULL OR a.publish_up <= ' . $nowDate . ')')
-						->where('(a.publish_down IS NULL OR a.publish_down >= ' . $nowDate . ')');
+					$nowDate = Factory::getDate()->toSql();
+
+					$query->extendWhere(
+						'AND',
+						['a.publish_up IS NULL',
```
Comment (Quy) [inline components/com_content/Model/ArticleModel.php]: Move to next line.

File: components/com_content/Model/ArticleModel.php [hunk 2]
```diff
@@ -119,50 +166,54 @@ public function getItem($pk = null)
...
-					$nowDate = $db->quote($date->toSql());
-
-					$query->where('(a.publish_up IS NULL OR a.publish_up <= ' . $nowDate . ')')
-						->where('(a.publish_down IS NULL OR a.publish_down >= ' . $nowDate . ')');
+					$nowDate = Factory::getDate()->toSql();
+
+					$query->extendWhere(
+						'AND',
+						[
+							'a.publish_up IS NULL',
```
Comment (alikon) [inline components/com_content/Model/ArticleModel.php]: add quoteName()
```suggestion
							$db->quoteName('a.publish_up') .' IS NULL',
							$db->quoteName('a.publish_up') . ' <= :publishUp',
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ec5e0cfdff5d41f21f73dc3f43fbcbde9e4d40cd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27721">issues.joomla.org/tracker/joomla-cms/27721</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on ec5e0cfdff5d41f21f73dc3f43fbcbde9e4d40cd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27721">issues.joomla.org/tracker/joomla-cms/27721</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27721">issues.joomla.org/tracker/joomla-cms/27721</a>.</sub>

**Pull request merged**